olp-cpp-sdk-core/src/client/PendingRequests.cpp

```diff
@@ -44,6 +44,10 @@ bool PendingRequests::CancelPendingRequests() {
     pair.second.cancel();
   }
 
+  for (auto context : contexts) {
+    context.CancelToken().cancel();
+  }
```

Comment:
But this is done in context.BlockingCancel() called below.
Why do we need it twice?

Reply:
Actually we need this code, we can't rely on the order of tasks in std::unorder_set.
Checked locally with performance tests

Reply:
Then better add in TaskContext::BlockingCancel() a guard to protect from double cancelling the CancellationContext:

```cpp
 bool BlockingCancel(std::chrono::milliseconds timeout) override {
      if (state_.load() == State::COMPLETED) {
        return true;
      }

      // Cancel operation and wait for notification
     if (!context_.IsCancelled()){
          context_.CancelOperation();
     }

      {
        std::lock_guard<std::mutex> lock(mutex_);
        execute_func_ = nullptr;
      }

      return condition_.Wait(timeout);
    }
```